utils/mil_tools/scripts/pythonFinder.py

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env python
+import os
+import re
+import sys
+import subprocess
+import shlex
+
+# Joseph Brooks
+# Checks NaviGator for all python files
+
+if len(sys.argv) == 1:
+    print("add directory path to be searched as an argument")
+    sys.exit()
+
+directory = sys.argv[1]
+
+allFileList = []
+
+# gets all files in directory
+for root, dirs, files in os.walk(directory):
+    for name in files:
+        programName = os.path.join(root, name)
+        allFileList.append(programName)
+
+pythonList = []
+
+# attempts to open file as text and checks #!/usr/bin/env
+# Just checks the first line
+for program in allFileList:
```

Comment:
You can also check if the file ends in .py in this loop so you don't need to worry about double entries